packages/common-lib/src/renderers/reports/sqr601.ts

```diff
@@ -178,41 +186,75 @@ export const sqr601DocumentDefinition = (
 					alignment: 'left',
 				},
 				{
-					// Eligible date for next promotion
+					// Eligible date for next promotion - or number of weeks since joined for c/ab
 					text: Maybe.isSome(loopmember.requirements.LastAprvDate)
-						? new Date(
-								loopmember.requirements.LastAprvDate.value +
-									60 * 60 * 24 * 56 * 1000,
-						  ).toLocaleDateString('en-US')
-						: '',
+						? loopmember.requirements.LastAprvDate.value -
+								new Date('01/01/2010').getTime() >
+						  0
+							? new Date(
+									loopmember.requirements.LastAprvDate.value +
+										60 * 60 * 24 * 56 * 1000,
+							  ).toLocaleDateString('en-US')
+							: Math.round(
+									Math.round(
+										new Date().getTime() -
+											new Date(loopmember.member.joined).getTime(),
+									) /
+										(1000 * 60 * 60 * 24 * 7),
+							  )
+						: Math.round(
+								Math.round(
+									new Date().getTime() -
+										new Date(loopmember.member.joined).getTime(),
+								) /
+									(1000 * 60 * 60 * 24 * 7),
+						  ),
 					fontSize: mySmallFontSize,
 					bold: false,
 					alignment: 'left',
 				},
 				{
 					// Next Grade
 					text:
-						CadetPromotionRequirementsMap[loopmember.requirements.NextCadetAchvID]
+						CadetPromotionRequirementsMap[loopmember.requirements.NextCadetGradeID]
 							.Grade,
 					fontSize: mySmallFontSize,
 					bold: false,
 					alighment: 'left',
 				},
+				{
+					// Current achievement
+					text:
+						loopmember.requirements.CurrentCadetAchv.CadetAchvID.toString() +
+						'-' +
+						loopmember.requirements.MaxAprvStatus.substr(0, 1),
+					fontSize: mySmallFontSize,
+					bold: false,
+					alighment: 'left',
+				},
 				{
 					// Lead Lab pass date or N/A if not required
 					text:
-						new Date(loopmember.requirements.CurrentCadetAchv.LeadLabDateP).getTime() -
-							new Date('01/01/2010').getTime() <=
-						0
+						loopmember.requirements.MaxAprvStatus === 'APR' // if approved status, display a blank or 'N/A' for next
+							? CadetPromotionRequirementsMap[loopmember.requirements.NextCadetAchvID]
+									.Leadership === 'None'
+								? 'N/A'
+								: ''
+							: // if incomplete or pending status, display current requirements (even if all filled in)
+							new Date(
+									loopmember.requirements.CurrentCadetAchv.LeadLabDateP,
+							  ).getTime() -
+									new Date('01/01/2010').getTime() <=
+							  0
 							? CadetPromotionRequirementsMap[loopmember.requirements.NextCadetAchvID]
 									.Leadership === 'None'
 								? 'N/A'
 								: ''
 							: new Date(
-									loopmember.requirements.CurrentCadetAchv.LeadLabDateP.substr(
-										0,
-										10,
-									),
+									loopmember.requirements.CurrentCadetAchv.LeadLabDateP.replace(
+										/-/g,
+										'/',
+									).replace(/T.+/, ''),
```

Comment:
Why the `.replace` call as opposed to the earlier `.substr` call?

Reply:
substr call resulted in locale dates, therefore one day different from CAPWATCH data.  replace forces zulu date and aligns with CAPWATCH dates.